source/sdk/structs/Stack.ooc

```diff
@@ -23,13 +23,16 @@ Stack: class <T> {
 				Exception new(This, "Trying to pop an empty stack.") throw()
 		this _data[--this _size]
 	}
+	top: func -> T {
+		this peek()
+	}
 	peek: func (index := 0) -> T {
 		version(safe)
 			if (index < 0)
 				Exception new(This, "Trying to peek(%d)! index must be >= 1 <= size" format(index)) throw()
 			else if (index >= this size)
 				Exception new(This, "Trying to peek(%d) a stack of size %d" format(index, this size)) throw()
-		this _data[index]
+		this _data[this size - index - 1]
 	}
 	//TODO: remove this function, exists only for compatibility with current codebase
 	empty?: func -> Bool {
```

Comment:
You only need to change one file (`Exception.ooc`) to get rid of this, please do.